### ISSUE TITLE
Exit pool rounding bug

### ISSUE THREAD

**Issue by alexcos20** [oceanprotocol/market#555]
**Describe the bug**

Looks like when using MAX to remove liquidity, sometimes we have rounding errors:

looking at the contract (https://etherscan.io/address/0x5e9939f6d959ffe9b328243dfadbeed9c46ac197#readContract), you will see that user 0x67e84944b441cb0a3fb0be4f98974d7363308b9d has 13219061360452614720 shares (wei)

but on exit, he tries to spend  13219061360452615000 pool shares  (2nd argument from https://etherscan.io/tx/0x981ae2085790239f0f45b430843ffdf1d2c28b7be1c62cd03dce2c42249ea612 .  which is 0xb7738a450ddebb58 (hex)

Balance:     13219061360452614720 
Requested: 13219061360452615000 

Clearly, looks like a rounding issue

**Comment by alexcos20** [comment]
another case:

User 0xa75776ce832a44d164160cb1ed1644aa0ae9a1fd  has  2256572329945371501 shares in pool [here](https://etherscan.io/address/0x2655b8a7357f4bb4a8cb2170e196096ac8f0cdf9#readContract)
![BPool _ 0x2655b8a7357f4bb4a8cb2170e196096ac8f0cdf9 - Google Chrome 4_25_2021 9_57_38 AM](https://user-images.githubusercontent.com/7489447/115983914-dddae400-a5ac-11eb-86ff-88ed716dacdf.png)

But when using MAX, he is trying to withdraw 2256572329945371700  (1f50f494085e9434 in hex, the 2nd parameter of the function)
![Ethereum Transaction Hash (Txhash) Details _ Etherscan - Google Chrome 4_25_2021 9_58_17 AM](https://user-images.githubusercontent.com/7489447/115983947-14186380-a5ad-11eb-93c0-3ca441f28861.png)


Shares: 2256572329945371501 
With:    2256572329945371700

**Comment by alexcos20** [comment]
Ideally, when using MAX, we should flag it and get the exact quantity from ocean.js before calling exitPool

**Comment by DimitarSD** [comment]
When we use MAX, we calculate the amount of pool shares:

`const amountPoolShares = (Number(amountMaxPercent) / 100) * Number(poolTokens)`

`poolTokens` (initially a string) is equal to `1.480962078984242346`, however, once we convert it to number (`Number(poolTokens)`) in the formula above we get `1.4809620789842424` - which is rounded to contains only 16 digits (2 digits less). We then have:

1.480962078984242346 - string
1.4809620789842424 - when converted to a Number

When we call `removePoolLiquidity` we pass the rounded version (2 digits less) which results in `1.480962078984242400` passed as an argument to exitPool after `parseFloat`

To summarize:
1.480962078984242346 - string
1.4809620789842424     - converted to a Number / passed to removePoolLiquidity
1480962078984242400 - passed to exitPool
